### ISSUE TITLE
./get-data-wiki.sh fr: template errors warnings

### ISSUE THREAD

**Issue by formiel** [facebookresearch/XLM#149]
Thank you so much for releasing the code! It is extremely helpful to me in understanding and implementing the state-of-the-art pretrained language models.

I would like to ask for your help related to the data preparation step. I'm trying to clean Wikipedia dump for French to train BERT from scratch and I had the following warnings when running the script `./get-data-wiki.sh fr`.

```
*** Cleaning and tokenizing fr Wikipedia dump ... ***
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
        LANGUAGE = (unset),
        LC_ALL = (unset),
        LC_CTYPE = "UTF-8",
        LANG = "fr_FR.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to a fallback locale ("fr_FR.UTF-8").
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
        LANGUAGE = (unset),
        LC_ALL = (unset),
        LC_CTYPE = "UTF-8",
        LANG = "fr_FR.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to a fallback locale ("fr_FR.UTF-8").
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
        LANGUAGE = (unset),
        LC_ALL = (unset),
        LC_CTYPE = "UTF-8",
        LANG = "fr_FR.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to a fallback locale ("fr_FR.UTF-8").
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
        LANGUAGE = (unset),
        LC_ALL = (unset),
        LC_CTYPE = "UTF-8",
        LANG = "fr_FR.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to a fallback locale ("fr_FR.UTF-8").
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
        LANGUAGE = (unset),
        LC_ALL = (unset),
        LC_CTYPE = "UTF-8",
        LANG = "fr_FR.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to a fallback locale ("fr_FR.UTF-8").
Tokenizer Version 1.1
Language: fr
Number of threads: 8
```
I would like to ask if it could cause any problem because I also got the following warnings related to template errors in articles later. Is this kind of problem normal when cleaning Wikipedia dataset? I checked a part of the resulting text and it seems to be fine but I would like to know why this error arises, whether or not it could affect the training corpus and how to prevent it if possible.

```
WARNING: Template errors in article 'Georges Bataille' (1239): title(1) recursion(0, 0, 0)
WARNING: Template errors in article 'Bruant ortolan' (5482): title(1) recursion(0, 0, 0)
WARNING: Template errors in article 'Coupe de l'America' (24404): title(1) recursion(0, 0, 0)
WARNING: Template errors in article 'Poissy' (28819): title(1) recursion(0, 0, 0)
WARNING: Template errors in article 'Corrosion aqueuse' (37336): title(1) recursion(0, 0, 0)
WARNING: Template errors in article 'Quinquennat (politique)' (45202): title(1) recursion(0, 0, 0)
WARNING: Template errors in article 'Crat\xe8re de Chicxulub' (89478): title(1) recursion(0, 0, 0)
WARNING: Template errors in article 'Courcy (Marne)' (97227): title(1) recursion(0, 0, 0)
WARNING: Template errors in article 'Prix Harvey (comics)' (139360): title(1) recursion(0, 0, 0)
WARNING: Template errors in article 'Phil\xe9mon (po\xe8te)' (148849): title(1) recursion(0, 0, 0)
...
```
The `locale` setting in my server is:
```
LANG=fr_FR.UTF-8
LANGUAGE=
LC_CTYPE=UTF-8
LC_NUMERIC="fr_FR.UTF-8"
LC_TIME="fr_FR.UTF-8"
LC_COLLATE="fr_FR.UTF-8"
LC_MONETARY="fr_FR.UTF-8"
LC_MESSAGES="fr_FR.UTF-8"
LC_PAPER="fr_FR.UTF-8"
LC_NAME="fr_FR.UTF-8"
LC_ADDRESS="fr_FR.UTF-8"
LC_TELEPHONE="fr_FR.UTF-8"
LC_MEASUREMENT="fr_FR.UTF-8"
LC_IDENTIFICATION="fr_FR.UTF-8"
LC_ALL=
```
and `locale.getpreferredencoding()` returned `'ANSI_X3.4-1968'`.

Could you please help me on this problem or point me to some sources where I can figure out myself? 

Thank you very much in advance for your kind help!

**Comment by formiel** [comment]
I have changed the `locale`settings so that the variables `LANGUAGE` and `LC_ALL` are not empty, hence there are no warnings related to `locale` anymore when running the script. However, the warnings of template errors still exist. Could you please help me explain if this type of errors would affect somehow to the resulting corpus obtained after cleaning?

Thank you very much for your time!

**Comment by aconneau** [comment]
@formiel : it seems like a problem related to MOSES? Could you try on a simple example to tokenize some French text file with MOSES and see whether you have an error there?

**Comment by formiel** [comment]
@aconneau Thank you so much for your reply! I tried using MOSES tokeniser in the same way like your suggestion on other French corpora such as Europarl, News Crawl, News Commentary, etc. but there is no such error. 

I want to compare the resulting text of some of the articles that received the template errors warnings (there are 139 such articles) with the actual articles in French Wikipedia to see if there is any loss in content. I don't know how to retrieve all the text of an article from a large text file so I used the simple command `grep 'some_phrases_in_actual_fr_wiki_article' fr.all`. The sample phrases that I checked exist in the file. So I'm not sure if it's normal to have that type of error and could we just skip it or not.

Thank you again for your help!
